conn.go

```diff
@@ -1430,6 +1525,24 @@ func (c *Conn) executeQuery(ctx context.Context, qry *Query) *Iter {
 	case *resultVoidFrame:
 		return &Iter{framer: framer}
 	case *resultRowsFrame:
+		if x.meta.newMetadataID != nil {
+			stmtCacheKey := c.session.stmtsLRU.keyFor(c.host.HostID(), c.currentKeyspace, qry.stmt)
+			inflight, ok := c.session.stmtsLRU.get(stmtCacheKey)
+			if !ok {
+				// We didn't find the stmt in the cache, so we just re-prepare it
+				return c.executeQuery(ctx, qry)
+			}
+
+			// Updating the result metadata id in prepared stmt
+			//
+			// If a RESULT/Rows message reports
+			//      changed resultset metadata with the Metadata_changed flag, the reported new
+			//      resultset metadata must be used in subsequent executions
+			inflight.preparedStatment.resultMetadataID = x.meta.newMetadataID
+			inflight.preparedStatment.response = x.meta
```

Comment:
Hmm I think there might be a synchronization issue here as well, other goroutines can be accessing this preparedstatement object at the same time. Ideally the prepared statement object should be immutable so we could just replace the entire object in the cache instead of trying to modify it. This would get rid of the synchronization issue.

Reply:
Could you please add some details here? On replacing the entire object you mean just replacing `inflight.preperadStatment`? Or replacing the whole `inflight` inside the cache? For the second variant, I could implement it the same way as handling of **UNPREPARED** response message is implemented.

Reply:
Doing it the same way as handling of `UNPREPARED` is not really ideal, it is just evicting the prepared statement from the cache and forcing the driver to re-prepare.

Replacing `inflight.preparedStatement` won't fix anything unless this field is made `atomic`, I was thinking of replacing the whole `inflight` object in the cache so we don't have to worry about synchronization or having to make things `atomic` (since the cache itself already does this). I think this can be done by creating a new `inflight` object (with a new copy of the `preparedStatement` object) and just calling `*preparedLRU.add` since it looks like it has "upsert" semantics.

Reply:
Oh, now I see. Indeed, we could just create a copy of the `inflight.preparedStatement`, update its metadata and **result_metadata_id**, and upsert the updated one to the cache. 
I'll update this one as well.

Reply:
I'd say create a copy of `inflight.preparedStatement` but also of the `inflight` object itself to ensure no other goroutine is reading the `inflight` object that we're modifying with the new `preparedStatement`

Reply:
And this one is added too